### PR TITLE
error handling for undefined

### PR DESCRIPTION
**Pull request by jchris**
this is to fix the dev wart 

{"module":"WriteQueueImpl","level":"error","error":"`undefined` is not supported by the IPLD Data Model and cannot be encoded","msg":"Error processing task"}

I tried to be smart about it

## Summary by CodeRabbit

- **New Features**
	- Enhanced document update processing with integrated data sanitization. Document values are now validated and cleaned automatically, ensuring proper handling of special cases (null, undefined, non-numeric values, or malformed arrays) with errors returned for invalid inputs.

- **Tests**
	- Expanded test coverage confirms the updated behavior for edge-case values during document updates, including null, undefined, NaN, Infinity, and arrays containing invalid values.

### CODE REVIEW

File: src/crdt-helpers.ts [hunk 1]
```diff
@@ -53,6 +53,35 @@ function toString<K extends IndexKeyType>(key: K, logger: Logger): string {
   }
 }
 
+export function sanitizeDocumentFields(obj: unknown): unknown {
+  if (Array.isArray(obj)) {
+    return obj.map((item: unknown) => {
+      if (typeof item === 'object' && item !== null) {
+        return sanitizeDocumentFields(item);
```
Comment (mabels) [inline src/crdt-helpers.ts:60]: why call with undefined

File: src/crdt-helpers.ts [hunk 2]
```diff
@@ -53,6 +53,35 @@ function toString<K extends IndexKeyType>(key: K, logger: Logger): string {
...
+      }
+      return item;
+    });
+  } else if (typeof obj === 'object' && obj !== null) {
+    const typedObj = obj as Record<string, unknown>;
```
Comment (mabels) [inline src/crdt-helpers.ts:65]: this is also entered if it is a class instances

File: src/crdt-helpers.ts [hunk 3]
```diff
@@ -53,6 +53,35 @@ function toString<K extends IndexKeyType>(key: K, logger: Logger): string {
...
+    const result: Record<string, unknown> = {};
+    for (const key in typedObj) {
+      if (Object.hasOwnProperty.call(typedObj, key)) {
+        const value = typedObj[key];
+        if (value === null || (!Number.isNaN(value) && value !== undefined)) {
```
Comment (mabels) [inline src/crdt-helpers.ts:70]: why filter something with NaN?

File: src/crdt-helpers.ts [hunk 4]
```diff
@@ -53,6 +53,35 @@ function toString<K extends IndexKeyType>(key: K, logger: Logger): string {
...
+          if (typeof value === 'object' && key[0] !== '_') {
```
Comment (mabels) [inline src/crdt-helpers.ts]: startsWith

File: src/crdt.ts [hunk 5]
```diff
@@ -104,6 +105,10 @@ export class CRDT<T extends DocTypes> {
   async bulk(updates: DocUpdate<T>[]): Promise<CRDTMeta> {
     await this.ready();
     const prevHead = [...this.clock.head];
+    updates = updates.map((dupdate: DocUpdate<T>) => ({
+      ...dupdate,
+      value: sanitizeDocumentFields(dupdate.value) as typeof dupdate.value
```
Comment (mabels) [inline src/crdt.ts]: as T or better sanitize gets generic

File: src/crdt-helpers.ts (hunk 1 above)
Comment (jchris) [reply src/crdt-helpers.ts:60]: we only want to clean field names not array slots

File: src/crdt-helpers.ts (hunk 2 above)
Comment (jchris) [reply src/crdt-helpers.ts:65]: I think that should be OK, we just want to clean out undefined fields

File: src/crdt-helpers.ts (hunk 3 above)
Comment (jchris) [reply src/crdt-helpers.ts:70]: you get `NaN` from `parseInt("foo")`
